pkg/deployment/deployment_test.go

```diff
@@ -802,19 +803,19 @@ func TestValidate(t *testing.T) {
 			builderNil:    true,
 			definitionNil: false,
 			apiClientNil:  false,
-			expectedError: "error: received nil ClusterDeployment builder",
+			expectedError: "error: received nil builder",
```

Comment:
I'm also okay with just fully removing `deployment_test.go`'s version of TestValidate as well.  This is "extra" now.

Reply:
Removed deployment TestValidate